### PR TITLE
Elide DG->FD project when not needed

### PR DESCRIPTION
**Pull request by nilsdeppe**
## Proposed changes

- Elides the project to FD when all neighbors are doing DG.
- Commit "Add support for receiving DG volume data" is in a separate PR
- Commits are marked as "fixup" because I need to squash them together to get them to compile. This is just easier to review.

### Code review checklist

- [ ] The code is documented and the documentation renders correctly. Run
  `make doc` to generate the documentation locally into `BUILD_DIR/docs/html`.
  Then open `index.html`.
- [ ] The code follows the stylistic and code quality guidelines listed in the
  [code review guide](https://spectre-code.org/code_review_guide.html).
- [ ] The PR lists upgrade instructions and is labeled `bugfix` or
  `new feature` if appropriate.

### CODE REVIEW

File: src/Evolution/DgSubcell/PrepareNeighborData.hpp [hunk 1]
```diff
@@ -16,95 +16,136 @@
...
  * The local maximum and minimum of the evolved variables is added to
  * `Tags::DataForRdmpTci` for the RDMP TCI. Then the
  * data needed by neighbor elements to do reconstruction on the FD grid is sent.
  * The data to be sent is computed in the mutator
- * `Metavariables::SubcellOptions::GhostDataToSlice`, which returns a
- * `Variables` of the tensors to slice and send to the neighbors. The main
- * reason for having the mutator `GhostDataToSlice` is to allow sending
- * primitive or characteristic variables for reconstruction.
+ * `Metavariables::SubcellOptions::GhostVariables`, which returns a
+ * `Variables` of the tensors send to the neighbors. The main reason for having
```
Comment (kidder) [inline src/Evolution/DgSubcell/PrepareNeighborData.hpp]: to send to

File: src/Evolution/Systems/GrMhd/ValenciaDivClean/Subcell/PrimitiveGhostData.hpp [hunk 2]
```diff
@@ -4,25 +4,21 @@
...
-template <size_t Dim>
-class Mesh;
 template <typename T>
 class Variables;
 /// \endcond
 
 namespace grmhd::ValenciaDivClean::subcell {
 /*!
- * \brief Computes the rest mass density \f$\rho\f$, electron fraction
+* \brief Computes the rest mass density \f$\rho\f$, electron fraction
```
Comment (kidder) [inline src/Evolution/Systems/GrMhd/ValenciaDivClean/Subcell/PrimitiveGhostData.hpp]: don't delete the space

**Review by kidder — CHANGES_REQUESTED** [review]
you can squash, after fixing the two documentation issues

**Comment by nilsdeppe** [comment]
Rebased, squashed, and squashed in the dox fixes. Thanks for the review!

**Review by kidder — APPROVED** [review]

**Pull request merged**